internal/testutil/pkgdata_test.go

```diff
@@ -0,0 +1,62 @@
+package testutil_test
+
+import (
+	"archive/tar"
+
+	_ "embed"
+
+	"github.com/canonical/chisel/internal/testutil"
+	"gopkg.in/check.v1"
+)
+
+type pkgdataSuite struct{}
+
+var _ = check.Suite(&pkgdataSuite{})
+
+var pkgdataTestEntries = []testutil.TarEntry{{
+	Header: tar.Header{Name: "./"},
+}, {
+	Header: tar.Header{Name: "./admin/", Mode: 00700, Uname: "admin"},
+}, {
+	Header:  tar.Header{Name: "./admin/password", Mode: 00600, Uname: "admin"},
+	Content: []byte("swordf1sh"),
+}, {
+	Header:  tar.Header{Name: "./admin/setpassword", Mode: 04711, Uname: "admin"},
+	Content: []byte{0x7f, 0x45, 0x4c, 0x46, 0x02, 0x01, 0x01},
+}, {
+	Header: tar.Header{Name: "./data/"},
+}, {
+	Header:  tar.Header{Name: "./data/invoice.txt"},
+	Content: []byte("$ 10"),
+}, {
+	Header: tar.Header{Name: "./data/logs/"},
+}, {
+	Header:  tar.Header{Name: "./data/logs/task.log"},
+	Content: []byte("starting\nfinished\n"),
+}, {
+	Header: tar.Header{Name: "./data/shared/", Mode: 02777},
+}, {
+	Header: tar.Header{Name: "./home/"},
+}, {
+	Header: tar.Header{Name: "./home/alice/", Uid: 1000},
+}, {
+	Header:  tar.Header{Name: "./home/alice/notes", Uid: 1000},
+	Content: []byte("check the cat"),
+}, {
+	Header: tar.Header{Name: "./home/bob/", Uname: "bob", Uid: 1001},
+}, {
+	Header:  tar.Header{Name: "./home/bob/task.sh", Mode: 00700, Uname: "bob", Uid: 1001},
+	Content: []byte("#!/bin/sh\n"),
+}, {
+	Header: tar.Header{Name: "./logs/", Linkname: "data/logs"},
+}}
+
+//go:embed testdata/pkgdata.deb
+var pkgdataExpectedDeb []byte
```

Comment:
Having the tests comparing against a pre-packaged binary means anyone testing this logic needs to unpack the deb, performs changes, repack the deb, and include the whole binary here. That's exactly what we need to do nowadays for the actual tests we have, so it feels like just moving the goalpost while preserving the same underlying issue.

How about just using our existing "tree-mapping" testing function to verify the output?  This is by the end of the day what the actual tests we have will compare the output against, so it feels realistic enough for our purposes, even more when what we have here is really a belt-and-suspender kind of approach in the sense we're testing the testing.

Reply:
@niemeyer I've removed the embedded deb file and now I'm checking tar entries against expected list. I didn't use the tree mapping code, because here, I can just check the list of entries coming out of tar Reader. IOW, I don't need to touch the file system to test this code.